uint32 neuron_index
float64 time